test/lib/rules/no-effect-override-spec.js

```diff
@@ -0,0 +1,74 @@
+"use strict"
+var rule = require("../../../lib/rules/no-effect-override")
+var RuleTester = require("eslint/lib/testers/rule-tester")
+var parserOptions = require("../../parserOptions")
+
+RuleTester.setDefaultConfig({
+  parserOptions: parserOptions
+})
+
+var ruleTester = new RuleTester()
+
+function buildTest(imports, code) {
+  var result = (imports && (imports + ";")) || ""
+  result += "function* test() { " + code + " }"
+  return result
+}
+
+ruleTester.run("no-effect-override", rule, {
+  valid: [
+    {
+      code: buildTest("import { take } from 'redux-saga'", "yield notAnEffect()")
+    },
+    {
+      code: buildTest("import { take as sagaTake } from 'redux-saga'", "const take = () => {}; yield sagaTake()")
+    },
+    {
+      code: buildTest("import { take as sagaTake } from 'redux-saga'", "function take() {}; yield sagaTake()")
```

Comment:
Semicolon is invalid after a function declaration (results in an extra empty statement following the declaration).